File: app/src/main/java/com/example/calculadora/MainActivity.java
package com.example.calculadora;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    TextView tvResultado;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText primer_numero = findViewById(R.id.et_num_uno);
        EditText segundo_numero = findViewById(R.id.et_num_dos);

        tvResultado = findViewById(R.id.tv_result);
        Button suma = findViewById(R.id.btn_suma);
        Button resta = findViewById(R.id.btn_resta);
        Button multiplicacion = findViewById(R.id.btn_multiplica);
        Button dividir = findViewById(R.id.btn_divide);
        Button salir = findViewById(R.id.btn_exit);

        clear(primer_numero, segundo_numero);

        primer_numero.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View view) { suma(primer_numero,segundo_numero);}});
        suma.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { suma(primer_numero,segundo_numero);}});

        resta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { resta(primer_numero,segundo_numero);}
        });

        multiplicacion.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { multiplicacion(primer_numero,segundo_numero);}
        });

        dividir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { divide(primer_numero,segundo_numero);}
        });
        salir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { finish();}
        });

    }
    private void divide(EditText primer_numero, EditText segundo_numero) {


        if(primer_numero.getText().toString().isEmpty() || segundo_numero.getText().toString().isEmpty() ) {
            error("los números no pueden ir vacios");
        }else if(segundo_numero.getText().toString().equals("0")){
            error("No se puede dividir por cero");
        }else{
            Integer primer = Integer.parseInt(primer_numero.getText().toString());
            Integer segundo = Integer.parseInt(segundo_numero.getText().toString());

            String res = String.valueOf((primer/segundo));
            mostrar(res, "división", primer_numero, segundo_numero);
        }
    }

    private void multiplicacion(EditText primer_numero, EditText segundo_numero) {
        if(!primer_numero.getText().toString().isEmpty()  && !segundo_numero.getText().toString().isEmpty()){
            Integer primer = Integer.parseInt(primer_numero.getText().toString());
            Integer segundo = Integer.parseInt(segundo_numero.getText().toString());

            String res = String.valueOf((primer * segundo));
            mostrar(res, "multiplicación", primer_numero, segundo_numero);
        }else{
            error("los números no pueden ir vacios");
        }
    }

    private void resta(EditText primer_numero, EditText segundo_numero) {
        if(!primer_numero.getText().toString().isEmpty()  && !segundo_numero.getText().toString().isEmpty()){
            Integer primer = Integer.parseInt(primer_numero.getText().toString());
            Integer segundo = Integer.parseInt(segundo_numero.getText().toString());

            String res = String.valueOf((primer - segundo));
            mostrar(res, "resta", primer_numero, segundo_numero);
        }else{
            error("los números no pueden ir vacios");
        }
    }

    private void suma(EditText primer_numero, EditText segundo_numero) {
        if(!primer_numero.getText().toString().isEmpty()  && !segundo_numero.getText().toString().isEmpty()){
            Integer primer = Integer.parseInt(primer_numero.getText().toString());
            Integer segundo = Integer.parseInt(segundo_numero.getText().toString());

            String res = String.valueOf((primer + segundo));
            mostrar(res, "suma", primer_numero, segundo_numero);
        }else{
            error("los números no pueden ir vacios");
        }
    }

    private void mostrar(String resultado,String funcion,EditText primer_numero, EditText segundo_numero){
        tvResultado.setText(resultado);
        tvResultado.setVisibility(View.VISIBLE);
        Toast.makeText(getBaseContext(), "El resultado de la "+funcion+" es "+resultado , Toast.LENGTH_LONG).show();
        clear(primer_numero, segundo_numero);
    }
    private void error(String errorText){
        Toast.makeText(getBaseContext(), errorText , Toast.LENGTH_LONG).show();
    }

    private void clear(EditText primer_numero, EditText segundo_numero){
        primer_numero.setText("");
        segundo_numero.setText("");
    }
}

File: app/src/main/java/com/example/calculadora/MainActivity.kt
package com.example.calculadora

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var tvResultado: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvResultado = findViewById(R.id.tv_result)

        //nombres de variables no pueden tener underscore
        val primerNumero = findViewById<EditText>(R.id.et_num_uno)
        val segundoNumero = findViewById<EditText>(R.id.et_num_dos)
        val suma = findViewById<Button>(R.id.btn_suma)
        val resta = findViewById<Button>(R.id.btn_resta)
        val multiplicacion = findViewById<Button>(R.id.btn_multiplica)
        val dividir = findViewById<Button>(R.id.btn_divide)
        val salir = findViewById<Button>(R.id.btn_exit)

        clear(primerNumero, segundoNumero)
        primerNumero.setOnClickListener { suma(primerNumero, segundoNumero) }
        suma.setOnClickListener { suma(primerNumero, segundoNumero) }
        resta.setOnClickListener { resta(primerNumero, segundoNumero) }
        multiplicacion.setOnClickListener { multiplicacion(primerNumero, segundoNumero) }
        dividir.setOnClickListener { divide(primerNumero, segundoNumero) }
        salir.setOnClickListener { finish() }
    }

    private fun divide(primer_numero: EditText, segundo_numero: EditText) {
        //negar un .isEmpty con exclamacion para que? si tenemos un  .isNotEmpty

        if (primer_numero.text.toString().isNotEmpty() || segundo_numero.text.toString().isNotEmpty()) {
            error("los números no pueden ir vacios")
        } else if (segundo_numero.text.toString() == "0") {
            error("No se puede dividir por cero")
        } else {
            //conversion directa a Double
            val primer = primer_numero.text.toString().toDouble()
            val segundo = segundo_numero.text.toString().toDouble()
            val res = (primer / segundo).toString()
            mostrar(res, "división", primer_numero, segundo_numero)
        }
    }

    private fun multiplicacion(primer_numero: EditText, segundo_numero: EditText) {
        //negar un .isEmpty con exclamacion para que? si tenemos un  .isNotEmpty
        if (primer_numero.text.toString().isNotEmpty() && segundo_numero.text.toString().isNotEmpty()) {
            val primer = primer_numero.text.toString().toInt()
            val segundo = segundo_numero.text.toString().toInt()
            val res = (primer * segundo).toString()
            mostrar(res, "multiplicación", primer_numero, segundo_numero)
        } else {
            error("los números no pueden ir vacios")
        }
    }

    private fun resta(primer_numero: EditText, segundo_numero: EditText) {
        if (primer_numero.text.toString().isNotEmpty() && segundo_numero.text.toString().isNotEmpty()) {
            val primer = primer_numero.text.toString().toInt()
            val segundo = segundo_numero.text.toString().toInt()
            val res = (primer - segundo).toString()
            mostrar(res, "resta", primer_numero, segundo_numero)
        } else {
            error("los números no pueden ir vacios")
        }
    }

    private fun suma(primer_numero: EditText, segundo_numero: EditText) {
        if (primer_numero.text.toString().isNotEmpty() && segundo_numero.text.toString().isNotEmpty()) {
            val primer = primer_numero.text.toString().toInt()
            val segundo = segundo_numero.text.toString().toInt()
            val res = (primer + segundo).toString()
            mostrar(res, "suma", primer_numero, segundo_numero)
        } else {
            error("los números no pueden ir vacios")
        }
    }

    private fun mostrar(
        resultado: String,
        funcion: String,
        primer_numero: EditText,
        segundo_numero: EditText
    ) {
        tvResultado!!.text = resultado
        tvResultado!!.visibility = View.VISIBLE
        Toast.makeText(baseContext, "El resultado de la $funcion es $resultado", Toast.LENGTH_LONG)
            .show()
        clear(primer_numero, segundo_numero)
    }

    private fun error(errorText: String) {
        Toast.makeText(baseContext, errorText, Toast.LENGTH_LONG).show()
    }

    private fun clear(primerNumero: EditText, segundoNumero: EditText) {
        primerNumero.setText("")
        segundoNumero.setText("")
    }
}